arangod/Transaction/Manager.cpp

```diff
@@ -1018,19 +1026,21 @@ Result Manager::statusChangeWithTimeout(TransactionId tid,
       // timeout
       break;
     }
-    std::this_thread::yield();
+    co_await sleepUsingSchedulerIfAvailable("trx-status-change-backup",
+                                            std::chrono::milliseconds{10});
   }
-  return res;
+  co_return res;
 }
 
-Result Manager::commitManagedTrx(TransactionId tid,
-                                 std::string const& database) {
+futures::Future<Result> Manager::commitManagedTrx(TransactionId tid,
+                                                  std::string const& database) {
   READ_LOCKER(guard, _hotbackupCommitLock);
-  return statusChangeWithTimeout(tid, database, transaction::Status::COMMITTED);
+  co_return co_await statusChangeWithTimeout(tid, database,
```

Comment:
Can the `_hotbackupCommitLock` now be held while the execution is suspendend, and then be woken up in a potentially different thread?

Reply:
```
basics::ReadWriteLock _hotbackupCommitLock;
```
afaik this is our own lock and is it capable of being released in a different thread. I will double check that.

Reply:
Yes, it is our own ReadWriteLock implementation, which supports locking/unlocking being performed by different threads.
The question was also more like: do we actually want to hold the hot backup lock here even if we cannot change the transaction's status and need to retry?
In this case, we could give up the hot backup lock temporarily and reacquire it later, when the operation resumes. Otherwise we would keep holding the hot backup lock for no good reason. 
Obviously the previous implementation already had the same issue, so this change would rather be a orthogonal and doesn't necessarily need to be done in this PR, if at all.